app/controllers/publishers/ats_api/v1/vacancies_controller.rb

```diff
@@ -1,58 +1,83 @@
 class Publishers::AtsApi::V1::VacanciesController < Api::ApplicationController
   before_action :authenticate_client!
 
+  rescue_from StandardError, with: :render_server_error
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+  rescue_from ActionController::ParameterMissing, with: :render_bad_request
+
   def index
-    @pagy, @vacancies = pagy(vacancies, items: 100)
+    @pagy, @vacancies = pagy(vacancies.where(publisher_ats_api_client: client), items: 100)
 
-    respond_to(&:json)
+    render json: {
+      data: @vacancies.map { |vacancy| render_vacancy(vacancy) },
+      meta: { totalPages: @pagy.pages, count: @pagy.page },
+    }
   end
 
   def show
-    @vacancy = vacancy
+    vacancy = Vacancy.find(params[:id])
+    render json: render_vacancy(vacancy)
   end
 
-  # No idea why rubocop can't see the check after the create call
-  # rubocop:disable Rails/SaveBang
   def create
-    @vacancy = Vacancy.create(vacancy_params)
-
-    respond_to do |format|
-      format.json do
-        if @vacancy.persisted?
-          render status: :created
-        else
-          render status: :bad_request
-        end
-      end
+    result = Publishers::AtsApi::V1::CreateVacancyService.new(permitted_vacancy_params).call
```

Comment:
don't really understand this 'create service object' pattern. Could it not just be a function call?

Reply:
I was meaning a 'function' rather than a class. I often see these 'function' objects become hard to change as all methods have access to the 'state' (which isn't real). IMHO this service object could do a little less (e.g. not return the specific controller status)